GitExtUtils/GitArgumentBuilder.cs

```diff
@@ -171,22 +98,40 @@ public void Add(GitConfigItem configItem)
             }
         }
 
-        /// <inheritdoc />
-        public override string ToString()
+        /// <summary>
+        /// Gets the length of the expected command.
+        /// </summary>
+        /// <returns>THe length of the expected command.</returns>
+        public override int GetLength()
         {
-            var arguments = base.ToString();
             var gitArgsLength = _gitArgs.Length;
             if (gitArgsLength == 0)
             {
                 gitArgsLength = -1; // prevent extra capacity when length is 0
             }
 
             // 7 = "-c " + "=" + " " + 2 (for possible quotes of value)
-            var capacity = _configItems.Sum(i => i.Key.Length + i.Value.Length + 7) + _command.Length + 1 + arguments.Length;
-            capacity += gitArgsLength + 1;
+            var length = _configItems.Sum(i => i.Key.Length + i.Value.Length + 7) +
+                         _command.Length + 1 +
+                         gitArgsLength + 1 +
+                         base.GetLength();
+
+            return length;
+        }
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var capacity = GetLength();
             var str = new StringBuilder(capacity);
 
+            var arguments = base.ToString();
+            var gitArgsLength = _gitArgs.Length;
+            if (gitArgsLength == 0)
+            {
+                gitArgsLength = -1; // prevent extra capacity when length is 0
+            }
+
```

Comment:
not needed anymore

Reply:
Why?

Reply:
Because it is used only once: in the next line `if (>0)`.

Reply:
It is actually needed.
We add 1 to `gitArgsLength` in the following line, so to not add if `gitArgsLength` is empty we take 1 here.

Reply:
Github has hidden too many lines. Look at the remaining function `ToString` (not at the factored out `GetLength`):
```
        public override string ToString()
        {
            var capacity = Length;
            var str = new StringBuilder(capacity);

            var arguments = base.ToString();
            var gitArgsLength = _gitArgs.Length;
            if (gitArgsLength == 0)
            {
                gitArgsLength = -1; // prevent extra capacity when length is 0
            }

            if (gitArgsLength > 0)
```
There's no `add 1`. The last line is the only usage. It can be replaced with `if (_gitArgs.Length) > 0`.